File: coffeetime-parent/RubyGun/src/04_data_type.rb
require 'bigdecimal'

# Everything is Object
# Data Type
x = 2
y = 0.1
flag = false
c = 'a'
foo = 'foo'

puts x.class
puts y.class
puts flag.class
puts true.class
puts c.class
puts foo.class
puts [].class

# BigDecimal
puts 2-1.8
puts BigDecimal.new('2')-BigDecimal.new('1.8')

# Type Conversion
puts 10.to_s.class

# Type Inference
if foo.is_a? String
  puts 'foo is a String'
end

File: coffeetime-parent/Python3Gun/src/04_data_type.py
import decimal

# Data Type
x = 2
y = 0.1
flag = False
c = 'a'
foo = 'foo'

print(type(x))
print(type(y))
print(type(flag))
print(type(True))
print(type(c))
print(type(foo))
print(type([]))
print(x.__class__)
print(1 == True)
print(True + True)

# Decimal
print(2 - 1.8)
print(decimal.Decimal('2') - decimal.Decimal('1.8'))

# Type Conversion
print(type(str(10)))
print(type(int('1')))

# Type Inference
if isinstance(foo, str):
    print('foo is a String')
